backend/src/main/java/com/soen341/instagram/service/impl/AccountService.java

```diff
@@ -0,0 +1,53 @@
+package com.soen341.instagram.service.impl;
+
+import java.util.Date;
+import java.util.regex.Pattern;
+
+// Spring
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import com.soen341.instagram.exception.account.InvalidEmailFormatException;
+
+// Project
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.model.Account;
+import com.soen341.instagram.utils.AccountVerifier;
+import com.soen341.instagram.utils.UserAccessor;
+
+@Service
+public class AccountService 
+{
+	@Autowired
+	AccountRepository accountRepository;
+
+  public void setPassword(final String newPassword) {
+      String oldPassword = UserAccessor.getCurrentAccount(accountRepository).getPassword();
```

Comment:
On line 25 and 27, you are using `UserAccessor.getCurrentAccount` twice, thus also querying the database twice for the same object. Alternatively, you could simple do line 27, and then grab the password through that `Account` object. This would reduce the need to query db too often cause it could get heavy.